parsers/IN_WE.py

```diff
@@ -93,17 +98,46 @@ def fetch_data(
                 item[datetime_col] = dt.shift(minutes=1).floor("minute").datetime
             else:
                 item[datetime_col] = dt.floor("minute").datetime
-        df_data = pd.DataFrame(
-            [item for item in data if item[datetime_col].hour == dt_12_hour.hour]
-        )
-        return df_data
+        return data
     else:
         raise ParserException(
             parser="IN_WE.py",
             message=f"{target_datetime}: {kind} data is not available",
         )
 
 
+def format_raw_data(
+    kind: str,
+    data: json,
+    zone_key: str,
+    target_datetime: datetime,
+    zone_key2: Optional[str] = None,
+) -> dict:
+    assert data is not None
```

Comment:
Nothing to change there, it's just a small comment.
I know we discussed whether we should include checks in our functions to make user users adhere to the function signature.

Usually functions should be self contained we should not assume other functions to take care of sanitization but it could be acceptable that since we supply type hints, if the user supplies a None they should expect stuff will go wrong, at least in other languages a None wouldn't be accepted :) So I would say it's also acceptable to not put the asserts.